ee/src/main/java/org/jboss/as/ee/component/ViewService.java

```diff
@@ -193,11 +195,24 @@ public Object invoke(InterceptorContext interceptorContext) throws Exception {
             if(component instanceof BasicComponent) {
                 ((BasicComponent) component).waitForComponentStart();
             }
-            final Method method = interceptorContext.getMethod();
+            final Method method = matchMethod(interceptorContext.getMethod());
             final Interceptor interceptor = viewInterceptors.get(method);
             return interceptor.processInvocation(interceptorContext);
         }
 
+        private Method matchMethod(Method invokedMethod) {
+            if (invokedMethod.getDeclaringClass().isInterface()) {
+                for (Method beanMethod : proxyFactory.getCachedMethods()) {
+                    if (MethodIdentifier.getIdentifierForMethod(beanMethod).equals(MethodIdentifier.getIdentifierForMethod(invokedMethod))) {
+                        return beanMethod;
+                    }
+                }
```

Comment:
This is on the request path and should be optimized to a single map lookup.

Reply:
I was referring to the entire for loop.

Reply:
This path is only executed when `invokedMethod.getDeclaringClass().isInterface()` is true, i.e., when the bean is 2.x ejb, or the ejb bean class has no default constructor. So for the majority of ejb use case, this part is not touched. So I updated this part to populate a map on-demand to save the mapping.

Reply:
Nice; I like that approach.